_posts/2016-03-01-Docker-meets-Continuous-Deployment.md

```diff
@@ -0,0 +1,165 @@
+
+Without the **artifact** we won't be able to build any image. We have to tell our build how to find **artifacts** from the **Build** phase. We can do that by defining an **Artifact Dependency** in **TeamCity**. We just have to choose a build configuration and define files which we want to use in this build.
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/release_artifacts.png" alt="Release Artifact Dependency" title="Release Artifact Dependency" class="img img-responsive style-screengrab">
+
+And finally we have to trigger this build automatically when we pass all **tests** in the previous step. By introducing **Finish Build Trigger** we can start this build just after **TeamCity** successfully finishes building the application.  
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/release_trigger.png" alt="Release Trigger" title="Release Trigger" class="img img-responsive style-screengrab">
+
+Now we are ready for release. Three build steps must be introduced: **Build Image**, **Push Image to Repository** and **Save version**. 
+This time we will use a different runner type: **Command Line**. We can execute a shell script on a build agent. Because we already installed **Docker** on our build agent we can use `docker` command in our shell script.
+ 
+### Build Image 
+
+To build the image we need to execute the following command: 
+
+```bash
+docker build --tag registry.private/simple_application:%build.number% .
+```
+Docker's build command will take our `Dockerfile` and build the image tagged as `registry.private/simple_application` and version `%build.number%`. 
+The variable `%build.number%` is a build-in **TeamCity** variable containing the current build number.
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/release_step_1.png" alt="Release Build Image Step" title="Release Build Image Step" class="img img-responsive style-screengrab">
+
+### Push Image
+
+The image created in the previous step exists only on the agent machine. To make the image available to others we need to store it in a repository. We can use **Docker Hub**, but in our example we use a private repository available under the address `repository.private`. We can execute the following command to push the image to the repository.
+
+```bash
+docker push registry.private/simple_application:%build.number%
+```
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/release_step_2.png" alt="Release Push Image to Repository Step" title="Release Push Image to Repository Step" class="img img-responsive style-screengrab">
+
+### Save Version
+
+The image is safely stored in the repository, but we need to do one more step: save the version of our image. Once again we run some shell script to generate an `image.version file: 
```

Comment:
Do you want to say "...save our docker image.." instead of ...save the version of our images... ?